clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5220,13 +5231,45 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       // Set O2 optimization level by default
       if (!Args.getLastArg(options::OPT_O_Group))
         CmdArgs.push_back("-O2");
+      // Add any predefined macros associated with intel_gpu* type targets
+      // passed in with -fsycl-targets
+      // TODO: Macros are populated during device compilations and saved for
+      // addition to the host compilation. There is no dependence connection
+      // between device and host where we should be able to use the offloading
+      // arch to add the macro to the host compile.
+      auto addTargetMacros = [&](const llvm::Triple &Triple) {
+        if (!Triple.isSPIR() && !Triple.isNVPTX() && !Triple.isAMDGCN())
```

Comment:
we probably want to use `isSPIROrSPIRV` everywhere, we are having some early-stage talks of maybe stoping using `spir64` because we don't actually generate SPIR 1.0, we generate SPIR-V, so it would be a lot easier if `spirv(32,64)` already worked :)